autogpt_platform/backend/test/load_store_agents.py

```diff
@@ -399,17 +406,25 @@ async def main():
                 continue
 
             metadata = csv_metadata[version_id]
-            print(f"\nProcessing: {metadata['agent_name']}")
-
-            # Load and create the agent graph
-            agent_data = await load_agent_json(json_file)
-            graph_id, graph_version = await create_agent_graph(
-                db, agent_data, known_blocks
-            )
-            loaded_graphs[graph_id] = (graph_id, graph_version)
-
-            # Create store listing
-            await create_store_listing(db, graph_id, graph_version, metadata)
+            agent_name = metadata["agent_name"]
+            print(f"\nProcessing: {agent_name}")
+
+            # Use a transaction per agent to prevent dangling resources
+            try:
+                async with db.tx() as tx:
+                    # Load and create the agent graph
+                    agent_data = await load_agent_json(json_file)
+                    graph_id, graph_version = await create_agent_graph(
+                        tx, agent_data, known_blocks
```

Comment:
I assume this is tested? we can drop-in replace db -> tx and it's going to work?